Library/Homebrew/extend/os/linux/hardware/cpu.rb

```diff
@@ -36,12 +39,10 @@ def family
             :haswell
           when 0x3d, 0x47, 0x4f, 0x56
             :broadwell
-          when 0x5e
+          when 0x4e, 0x55, 0x5e, 0x8e, 0x9e
             :skylake
-          when 0x8e
-            :kabylake
           else
-            :dunno
```

Comment:
`:dunno` has a special meaning elsewhere in the code; is `:unknown_` going to be handled correctly with those?

Reply:
https://github.com/Homebrew/brew/search?q=dunno
I see special treatment of `:dunno` here:
https://github.com/Homebrew/brew/blob/4897ec02d0f00c37a21ddaa0bebff6b83b27d027/Library/Homebrew/hardware.rb#L14
which seems unnecessary since `-march=native` is the default. I don't see any other special treatment of `:dunno`. Do you know of a case that you can point me to?

Reply:
> which seems unnecessary since `-march=native` is the default

Only for `clang`, it seems. I think it'd be maybe worth handling the `:unknown*` case specifically in `optimization_flags`?

Reply:
I've changed the default optimization flags to `-march=native` in commit https://github.com/Homebrew/brew/pull/5012/commits/440a5e3e27aa39b7495d187be647caf4f98ffc73.

`optimization_flags.fetch` is called in three places:
https://github.com/Homebrew/brew/blob/f354160c7179ff0a615f2023393213736f8d6191/Library/Homebrew/extend/ENV/super.rb#L233-L237
https://github.com/Homebrew/brew/blob/f354160c7179ff0a615f2023393213736f8d6191/Library/Homebrew/extend/ENV/std.rb#L210
In the first two cases, `bottle_arch` and `oldest_cpu` are guaranteed to be in `optimization_flags`. In the last case, a default value of `DEFAULT_FLAGS` is provided.